spec/features/user_profile_spec.rb

```diff
@@ -0,0 +1,72 @@
+require 'rails_helper'
+
+RSpec.feature 'User profile', type: :feature, js: true do
+  let!(:user) { FactoryBot.create(:user) }
+  let!(:first_event) { FactoryBot.create(:event, user: user, finish_date: '31/12/2018', start_date: '30/12/2018') }
+  let!(:second_event) { FactoryBot.create(:event, user: user, finish_date: '01/01/2019', start_date: '31/12/2018') }
+  let!(:third_event) { FactoryBot.create(:event, user: user, finish_date: '21/12/2018', start_date: '20/12/2018') }
+  let!(:target_to_first_event) { FactoryBot.create(:target, event: first_event) }
+  let!(:target_to_second_event) { FactoryBot.create(:target, event: second_event) }
+  before(:each) do
+    sign_in(user)
+    visit user_path(id: user.id)
+  end
+
+  it 'show user data' do
+    expect(page).to have_content user.first_name
+    expect(page).to have_content user.last_name
+    expect(page).to have_content user.phone
+    expect(page).to have_content user.project.name
+    expect(page).to have_content user.project.desc
+  end
+
+  it 'show sorted event' do
+    finish_date_of_events_from_page = page.all('.event-finish-date').map(&:text)
+    expected_finish_date_list = [first_event.finish_date.iso8601,
+                                 second_event.finish_date.iso8601,
+                                 third_event.finish_date.iso8601].sort!.reverse
+
+    expect(page).to have_content first_event.description
+    expect(page).to have_content second_event.description
+    expect(page).to have_content third_event.status
+    expect(page).to have_content first_event.finish_date.year
+    expect(page).to have_content second_event.finish_date.year
+
+    expect(page).to_not have_content first_event.start_date
+    expect(page).to_not have_content first_event.comments
+    expect(page).to_not have_content first_event.summary
+
+    expect(expected_finish_date_list).to eq(finish_date_of_events_from_page)
+  end
+
+  it 'show hidden event data when click on event header' do
```

Comment:
Did not get this test. You clicked on first event, but you check if first and second event info  present.